### PR TITLE
Fix DDox exclusion for ObjC

### PR DESCRIPTION
**Pull request by wilzbach**
Required for https://github.com/dlang/dmd/pull/7513

### CODE REVIEW

**Comment by jacob-carlborg** [comment]
@wilzbach seems this didn't fix the issue.

**Comment by wilzbach** [comment]
> @wilzbach seems this didn't fix the issue.

So the problem was that we still use `ddmd` for the latest release, hence `src/dmd` is never found for the latest release.

**Comment by wilzbach** [comment]
Oh and we clone `dmd-2.077.1` _after_ we do the "ifeq" check, hence I had to move the checks into the target.

File: posix.mak [hunk 1]
```diff
@@ -725,16 +725,14 @@ else
...
-
 $G/docs-latest.json : ${DMD_LATEST} ${DMD_LATEST_DIR} \
 			${DRUNTIME_LATEST_DIR} ${PHOBOS_LATEST_FILES_GENERATED} | dpl-docs
+	# remove this after https://github.com/dlang/dmd/pull/7513 has been merged
+	if [ -f $(DMD_LATEST_DIR)/src/*/objc_glue_stubs.d ] ; then \
+	   DMD_EXCLUDE_LATEST="-e /objc_glue.d/d"; \
+	fi; \
 	find ${DMD_LATEST_DIR}/src -name '*.d' | \
-		sed -e /mscoff/d ${DMD_EXCLUDE_LATEST} ${DMD_EXCLUDE} \
+		sed -e /mscoff/d $${DMD_EXCLUDE_LATEST} ${DMD_EXCLUDE} \
```
Comment (jacob-carlborg) [inline posix.mak]: Should there be two `$`?

File: posix.mak [hunk 2]
```diff
@@ -747,19 +745,17 @@ $G/docs-latest.json : ${DMD_LATEST} ${DMD_LATEST_DIR} \
...
-   DMD_EXCLUDE_PRERELEASE += -e /objc_glue.d/d
-endif
-
 # DDox tries to generate the docs for all `.d` files. However for dmd this is tricky,
 # because the `{mach, elf, mscoff}` are platform dependent.
 # Thus the need to exclude these files.
 $G/docs-prerelease.json : ${DMD} ${DMD_DIR} ${DRUNTIME_DIR} \
 		${PHOBOS_FILES_GENERATED} | dpl-docs
+	# remove this after https://github.com/dlang/dmd/pull/7513 has been merged
+	if [ -f $(DMD)/src/*/objc_glue_stubs.d ] ; then \
```
Comment (jacob-carlborg) [inline posix.mak]: This should be `$(DMD_DIR)` not `$(DMD)`.

File: posix.mak (hunk 1 above)
Comment (wilzbach) [reply posix.mak]: Yes it's now a shell variable and `$$` escapes the `$`.

Comment (jacob-carlborg) [reply posix.mak]: Ok.

**Comment by jacob-carlborg** [comment]
😃.

**Comment by jacob-carlborg** [comment]
Can we get this merged?

**Comment by wilzbach** [comment]
> Note: you can test for file existence in make:

Hehe, I have been there, but this doesn't work for `DMD_LATEST` as the file _never_ exists - it's cloned later. Hence, the need for doing this in bash.

**Pull request merged**